lib/Sema/TypeCheckDecl.cpp

```diff
@@ -482,6 +482,53 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  if (!VD->getType()->isEqual(DC->getSelfTypeInContext()))
```

Comment:
Can you leave some comments around the chunks of this code to ensure someone looking here later knows why this check exists?